### PR TITLE
`Development`: Reduce number of test users in StudentExamIntegrationTest

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] The changed server tests succeed **locally** and on **Bamboo** and on **GitHub Actions**.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
In StudentExamIntegrationTest, there were multiple places where more than the actual needed amount of exams were created and prepared for conduction. By reducing the number of students and adapting the number of prepared exams for conduction, the tests do not only save memory but also time.

Locally, the time was reduced from ~38 sec to ~34 sec (might vary dependent on hardware).

On Bamboo, the tests are around 14 seconds faster:
[Develop, Bamboo Logs](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS-JAVATEST-4854/log): 1:17min
first test:	`17:35:17 StudentExamIntegrationTest > testDeleteExamWithMultipleTestRuns() STANDARD_OUT`
last test:	`17:36:34 StudentExamIntegrationTest > ChangedAndUnchangedSubmissionsIntegrationTest > testChangedAndNotSubmittedDragAndDropQuestionSubmission() PASSED`
[This branch, Bamboo Logs](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5280-JAVATEST-4/log): 1:03min
first test:	`18:16:11 StudentExamIntegrationTest > testDeleteExamWithMultipleTestRuns() STANDARD_OUT`
last test:	`18:17:14 StudentExamIntegrationTest > ChangedAndUnchangedSubmissionsIntegrationTest > testChangedAndNotSubmittedDragAndDropQuestionSubmission() PASSED`

### Description
In this PR, the  `prepareStudentExamsForConduction()` method was adapted. It now expects the number of prepared student exams as an input. By doing so, the number of prepared exams can be set individually for each test and calling the method does no longer create and prepare 5 student exams by default.

This is especially useful in places where only one StudentExam is needed (e.g. testStudentExamSummaryAsStudentBeforePublishResults_doFilter):
Before - created and prepared 5 StudentExams, but only needed one
`StudentExam studentExam = prepareStudentExamsForConduction(false, true).get(0);`
Now - create and prepare only 1 StudentExam
`StudentExam studentExam = prepareStudentExamsForConduction(false, true, 1).get(0);`

### Steps for Testing
**Test changes only**

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java [hunk 1]
```diff
@@ -173,12 +173,12 @@ class StudentExamIntegrationTest extends AbstractSpringIntegrationBambooBitbucke
...
     private final List<LocalRepository> studentRepos = new ArrayList<>();
 
-    private static final int NUMBER_OF_STUDENTS = 5;
+    private static final int NUMBER_OF_STUDENTS = 2;
 
     @BeforeEach
     void initTestCase() throws Exception {
-        // the service already creates 5 students, 1 tutor, 1 editor, 1 instructor
-        programmingExerciseTestService.setupTestUsers(TEST_PREFIX, 0, 0, 0, 1);
+        userUtilService.addUsers(TEST_PREFIX, 3, 1, 0, 2);
```
Comment (reschandreas) [inline src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java:180]: The second argument is called `numberOfStudents` is there a reason why you set it to 3 and not used the constant `NUMBER_OF_STUDENTS`?

**Review by reschandreas — CHANGES_REQUESTED** [review]
I have one question (see inline comment), otherwise, code LGTM 👍🏻 Thanks!

File: src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java (hunk 1 above)
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java:180]: Did you consider also reducing the number of instructors? Only the test `testFindAllTestRunsForExam()` needs the instructor2. You could consider adding the second instructor only in the beginning of this test

**Review by laadvo — COMMENTED** [review]
Changes look good, I just have a small suggestion. I agree with @reschandreas, you should use the constant `NUMBER_OF_STUDENTS` when adding users in the `initTestCase()`

File: src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java (hunk 1 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java]: There is a story behind this ^^ :

One of the tests previously needed a third student and therefor, I have set up the users with  `userUtilService.addUsers(TEST_PREFIX, NUMBER_OF_STUDENTS, 1, 0, 2);` and I have created the third user only for the one test that needed it. But after some testing, I realised that the additional creation of the third user was costing more time than I saved by reducing the amount of created students in the initTestCase() method. Therefor, I changed the number to 3, such that three students were created for each test and it became faster :O

Anyway, some commits later, I changed the test such that it does not require 3 students anymore. Looks like I forgot to undo my changes in the initTestCase() Method. I updated it just now :)

Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java]: Adding users only for individual tests is slower than creating the user in the initTestCase method. Seems a bit weird, but I tested it and it works like that on my machine xD
The number of students was updated 🚀

**Review by reschandreas — APPROVED** [review]
Changes now LGTM 👍🏻

File: src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java (hunk 1 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exam/StudentExamIntegrationTest.java]: Oh I see, in this case it makes sense to create two in the before each method. Interesting behavior :sweat_smile:

**Review by laadvo — APPROVED** [review]
Code looks good to me :+1:

**Pull request merged**